### PR TITLE
Add hubType to HubInvocationContext

### PR DESCRIPTION
**Pull request by Potapy4**
Summary of the changes (Less than 80 chars)
 - Just added the `HubName` property to the `HubInvocationContext`

Addresses #14344

/cc: @anurse

### CODE REVIEW

File: src/SignalR/server/Core/src/HubInvocationContext.cs
Comment (BrennanConroy) [inline src/SignalR/server/Core/src/HubInvocationContext.cs]: We should make this constructor call the other one (or vice versa)

**Review by BrennanConroy — COMMENTED** [review]
Good start!

Could you also update our usage of this type at https://github.com/aspnet/AspNetCore/blob/23596349092f69890c3efe5b8b9005856cc83db2/src/SignalR/server/Core/src/Internal/DefaultHubDispatcher.cs#L497

And update the test that uses this https://github.com/aspnet/AspNetCore/blob/23596349092f69890c3efe5b8b9005856cc83db2/src/SignalR/server/SignalR/test/HubConnectionHandlerTests.cs#L2218

**Comment by BrennanConroy** [comment]
Friendly ping @Potapy4

**Comment by Potapy4** [comment]
> Friendly ping @Potapy4

@BrennanConroy thanks for your review and reminder! 🎉 I pushed updates 👀

File: src/SignalR/server/Core/src/HubInvocationContext.cs
Comment (halter73) [inline src/SignalR/server/Core/src/HubInvocationContext.cs]:
```suggestion
        public Type Hub { get; }
```

I should have said something about this earlier, but I think we should expose the Hub type instead of just the Hub name. That way developers can get the fully qualified name and other details. Thanks.

File: src/SignalR/server/Core/src/Internal/DefaultHubDispatcher.cs
Comment (BrennanConroy) [inline src/SignalR/server/Core/src/Internal/DefaultHubDispatcher.cs]: nit: move this into the `IsHubMethodAuthorized` method after the `policies.Any()` check

File: src/SignalR/server/Core/src/HubInvocationContext.cs
Comment (Potapy4) [reply src/SignalR/server/Core/src/HubInvocationContext.cs]: 🤔 good point, but how do we get the name of the hub with `HubType` prop?

Comment (BrennanConroy) [reply src/SignalR/server/Core/src/HubInvocationContext.cs]: [Type.FullName](https://docs.microsoft.com/dotnet/api/system.type.fullname?view=netframework-4.8) or [Type.Name](https://docs.microsoft.com/dotnet/api/system.type.name?view=netstandard-1.6&viewFallbackFrom=netframework-4.8)

Comment (Potapy4) [reply src/SignalR/server/Core/src/HubInvocationContext.cs]: make sense, @BrennanConroy so should I change the `HubName` to `HubType` and revert my previous changes?

Comment (BrennanConroy) [reply src/SignalR/server/Core/src/HubInvocationContext.cs]: > should I change the `HubName` to `HubType`

Yes please!

Comment (BrennanConroy) [reply src/SignalR/server/Core/src/HubInvocationContext.cs]: @Potapy4 I think we're good to merge this PR once this change is in!

Comment (BrennanConroy) [reply src/SignalR/server/Core/src/HubInvocationContext.cs]: Oh, and since this is an API change, could you update the reference assemblies: https://github.com/aspnet/AspNetCore/blob/master/docs/ReferenceAssemblies.md#when-changing-public-api

Should just need to run `dotnet msbuild /t:GenerateReferenceSource` from the signalr/server/core/src/ folder.

Comment (halter73) [reply src/SignalR/server/Core/src/HubInvocationContext.cs]: @davidfowl @anurse Do we need any sort of API review before merging this?

**Comment by BrennanConroy** [comment]
@aspnet/asp-net-api-reviews could we get a quick review please? This adds a public constructor and property to the `HubInvocationContext` class.

File: src/SignalR/server/Core/src/HubInvocationContext.cs
Comment (Potapy4) [reply src/SignalR/server/Core/src/HubInvocationContext.cs]: > @Potapy4 I think we're good to merge this PR once this change is in!

Sorry for the delay, I will do it soon! Quick question: if we change it to `Type HubType` - should we still pass `hubname` to the constructor?

Comment (BrennanConroy) [reply src/SignalR/server/Core/src/HubInvocationContext.cs]: We would want to pass `hubType` in the constructor instead :)

**Review by davidfowl — APPROVED** [review]
API approved

**Review by BrennanConroy — APPROVED** [review]

**Comment by BrennanConroy** [comment]
Thanks @Potapy4 !

**Pull request merged**
